app/controllers/submissions_controller.rb

```diff
@@ -473,23 +499,36 @@ def turnitin_report
       end
     end
   end
+  private :plagiarism_report
 
-  def resubmit_to_turnitin
+  def resubmit_to_plagiarism(type)
     return render(:nothing => true, :status => 400) unless params_are_integers?(:assignment_id, :submission_id)
 
     if authorized_action(@context, @current_user, [:manage_grades, :view_all_grades])
       @assignment = @context.assignments.active.find(params[:assignment_id])
       @submission = @assignment.submissions.where(user_id: params[:submission_id]).first
-      @submission.resubmit_to_turnitin
+
+      if type == 'vericite'
+        # VeriCite
+        @submission.resubmit_to_vericite
+        messageKey = "resubmitted_to_vericite"
+        messageDefault = "Successfully resubmitted to VeriCite."
+      else
+        # turnitin
+        @submission.resubmit_to_turnitin
+        messageKey = "resubmitted_to_turnitin"
+        messageDefault = "Successfully resubmitted to turnitin."
+      end
       respond_to do |format|
         format.html {
-          flash[:notice] = t('resubmitted_to_turnitin', "Successfully resubmitted to turnitin.")
+          flash[:notice] = t(messageKey, messageDefault)
```

Comment:
@jenseng Thanks! That is good to know. I've made these changes and updated the PR.